packages/zwave-js/src/lib/commandclass/ThermostatFanModeCC.ts

```diff
@@ -0,0 +1,390 @@
+import type {
+	Maybe,
+	MessageOrCCLogEntry,
+	MessageRecord,
+	ValueID,
+} from "@zwave-js/core";
+import {
+	CommandClasses,
+	enumValuesToMetadataStates,
+	parseBitMask,
+	validatePayload,
+	ValueMetadata,
+	ZWaveError,
+	ZWaveErrorCodes,
+} from "@zwave-js/core";
+import { getEnumMemberName, pick } from "@zwave-js/shared";
+import type { Driver } from "../driver/Driver";
+import { MessagePriority } from "../message/Constants";
+import {
+	CCAPI,
+	PollValueImplementation,
+	POLL_VALUE,
+	SetValueImplementation,
+	SET_VALUE,
+	throwUnsupportedProperty,
+	throwWrongValueType,
+} from "./API";
+import {
+	API,
+	CCCommand,
+	CCCommandOptions,
+	ccValue,
+	ccValueMetadata,
+	CommandClass,
+	commandClass,
+	CommandClassDeserializationOptions,
+	expectedCCResponse,
+	gotDeserializationOptions,
+	implementedVersion,
+} from "./CommandClass";
+import { ThermostatModeCCGet } from "./ThermostatModeCC";
+
+// All the supported commands
+export enum ThermostatFanModeCommand {
+	Set = 0x01,
+	Get = 0x02,
+	Report = 0x03,
+	SupportedGet = 0x04,
+	SupportedReport = 0x05,
+}
+
+/**
+ * @publicAPI
+ */
+export enum ThermostatFanMode {
+	"Auto low" = 0x00,
+	"Low" = 0x01,
+	"Auto high" = 0x02,
+	"High" = 0x03,
+	"Auto medium" = 0x04,
+	"Medium" = 0x05,
+	"Circulation" = 0x06,
+	"Humidity circulation" = 0x07,
+	"Left and right" = 0x08,
+	"Up and down" = 0x09,
+	"Quiet" = 0x0a,
+	"External circulation" = 0x0b,
+}
+
+@API(CommandClasses["Thermostat Fan Mode"])
+export class ThermostatFanModeCCAPI extends CCAPI {
+	public supportsCommand(cmd: ThermostatFanModeCommand): Maybe<boolean> {
+		switch (cmd) {
+			case ThermostatFanModeCommand.Get:
+			case ThermostatFanModeCommand.SupportedGet:
+				return this.isSinglecast();
+			case ThermostatFanModeCommand.Set:
+				return true; // This is mandatory
+		}
+		return super.supportsCommand(cmd);
+	}
+
+	protected [SET_VALUE]: SetValueImplementation = async (
+		{ property },
+		value,
+	): Promise<void> => {
+		if (property !== "mode") {
+			throwUnsupportedProperty(this.ccId, property);
+		}
+		if (typeof value !== "number") {
+			throwWrongValueType(this.ccId, property, "number", typeof value);
+		}
+		await this.set(value);
+
+		if (this.isSinglecast()) {
+			// Verify the current value after a delay
+			this.schedulePoll({ property });
+		}
```

Comment:
This should also allow setting the `"off"` property. In that case, I'd load the current mode from the value DB and use it together with `off`.

Reply:
I just reread the specs on the Set command, and you're of course correct. I do think they really could have written that paragraph better.

Reply:
Sounds familiar 😅